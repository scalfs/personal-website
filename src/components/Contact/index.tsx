import Button from 'components/Button'
import LayoutBlur from 'components/LayoutBlur'
import Title from 'components/Title'
import { useSR } from 'hooks'

const Contact = () => {
  const refToContainer = useSR()

  return (
    <section
      id="contact"
      ref={refToContainer}
      className="relative mx-auto max-w-xl py-28 text-center"
    >
      <LayoutBlur type="lines" />

      <Title>Get in Touch</Title>

      <p>
        Right now I am looking for new opportunities. So, if you want to discuss
        a project, ask any question or just want to say hi, feel free to reach
        me. I&apos;ll be glad to get back at you as soon as possible.
      </p>

      <Button
        className="mx-auto mt-8 block w-fit"
        href="mailto:hello@vitorescalfoni.com"
      >
        Say Hello
      </Button>
    </section>
  )
}

export default Contact
