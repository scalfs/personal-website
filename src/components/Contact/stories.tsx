import { ComponentMeta, ComponentStory } from '@storybook/react'
import Contact from '.'

export default {
  title: 'Contact',
  component: Contact
} as ComponentMeta<typeof Contact>

const Story: ComponentStory<typeof Contact> = () => <Contact />

export const Default = Story.bind({})
