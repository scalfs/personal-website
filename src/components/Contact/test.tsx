import { render, screen } from '@testing-library/react'

import Contact from '.'

describe('<Contact />', () => {
  it('should render the section', () => {
    const { container } = render(<Contact />)

    expect(
      screen.getByRole('heading', { name: /Get in Touch/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('link', { name: /Say Hello/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
