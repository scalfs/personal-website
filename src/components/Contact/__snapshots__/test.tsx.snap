// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact /> should render the section 1`] = `
<section
  class="relative mx-auto max-w-xl py-28 text-center"
  id="contact"
>
  <div
    aria-hidden="true"
  >
    <div
      class="absolute bottom-1/3 right-0 h-1 w-64 -rotate-45 translate-x-40 bg-sky-500 blur-lg"
    />
    <div
      class="absolute bottom-1/3 right-16 h-1 w-64 -rotate-45 translate-x-40 bg-blue-500 blur-lg"
    />
    <div
      class="absolute bottom-1/3 right-28 h-1 w-64 -rotate-45 translate-x-40 bg-blue-800 blur-lg"
    />
  </div>
  <h1
    class="mb-4 text-4xl font-extrabold tracking-tight sm:mb-8 sm:text-5xl lg:text-6xl"
  >
    Get in Touch
  </h1>
  <p>
    Right now I am looking for new opportunities. So, if you want to discuss a project, ask any question or just want to say hi, feel free to reach me. I'll be glad to get back at you as soon as possible.
  </p>
  <a
    class="rounded-lg border border-slate-300 px-5 py-3 font-semibold text-blue-700 transition-colors duration-250 ease-in-out-cubic hover:border-blue-700 hover:bg-blue-100 focus:outline-none focus:ring-2 focus:ring-blue-700 focus:ring-offset-2 focus:ring-offset-slate-50 mx-auto mt-8 block w-fit"
    href="mailto:hello@vitorescalfoni.com"
    target="_blank"
  >
    Say Hello
  </a>
</section>
`;
